### PR TITLE
Docs: add migration guide for addon creators and addon consumers about the changes in addon APIs

### PR DESCRIPTION
**Pull request by ndelangen**
I added a ton of migration guide documentation for both addon creators and addon consumers

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -494,6 +497,112 @@ The new CLI commands remove the following flags:
...
+```js
+import { addons } from '@storybook/manager-api';
+
+addons.addPanel('my-panel', {
+  title: 'My Panel',
+  render: ({ active, key }) => <div>My Panel</div>,
+});
+```
+
+Not that this before would import `addons` from `@storybook/addons`, but now it imports `{ addons }` from `@storybook/manager-api`.
```
Comment (shilman) [inline MIGRATION.md:516]:
```suggestion
Note that this before would import `addons` from `@storybook/addons`, but now it imports `{ addons }` from `@storybook/manager-api`.
```

File: MIGRATION.md [hunk 2]
```diff
@@ -494,6 +497,112 @@ The new CLI commands remove the following flags:
...
+The `addons` export is now a named export only, there's no default export anymore, so make sure to update this usage.
+
+The package `@storybook/addons` is still available, but it's only for backwards compatibility. It's not recommended to use it anymore.
+
+It's also been used by addon creators to gain access to a few APIs like `makeDecorator`. 
+These APIs are now available in `@storybook/preview-api`.
+
+Storybook users have had access to a few storybook-lifecycle hooks such as `useChannel`, `useParameter`, `useStorybookState`;
+when these hooks are used in panels, they should be imported from `@storybook/manger-api`.
+when these hooks are used in decorators/stories, they should be imported from `@storybook/preview-api`.
```
Comment (shilman) [inline MIGRATION.md:526]:
```suggestion
When these hooks are used in decorators/stories, they should be imported from `@storybook/preview-api`.
```

File: MIGRATION.md [hunk 3]
```diff
@@ -494,6 +497,112 @@ The new CLI commands remove the following flags:
...
+
+Storybook will automatically detect if you're using the old API, and will convert it to the new API for you.
```
Comment (shilman) [inline MIGRATION.md:528]:
```suggestion
Storybook 7 includes `@storybook/addons` shim package that provides the old API and calls the new API under the hood.
```

File: MIGRATION.md [hunk 4]
```diff
@@ -494,6 +497,112 @@ The new CLI commands remove the following flags:
...
+
+If you're an addon creator, you'll have to update your addon to use the new APIs.
+
+That means you'll have to release a breaking release of your addon to make it compatible with Storybook 7.
+It should no longer depend on `@storybook/addons`, but instead on `@storybook/preview-api` and/or `@storybook/manager-api`.
+
+You might also depend (and use) these packages in your addon's decorators: `@storybook/store`, `@storybook/preview-web`, `@storybook/core-client`, `@storybook/client-api`; these have all been consolidated into `@storybook/preview-api`.
+So if you use any of these packages, please import what you need from `@storybook/preview-api` instead.
+
+If your addon is adding panels/etc. into storybook's manager UI, please be aware that storybook will optimize this to ensure no duplicate packages are loaded.
```
Comment (shilman) [inline MIGRATION.md:559]:
```suggestion
```

File: MIGRATION.md [hunk 5]
```diff
@@ -494,6 +497,112 @@ The new CLI commands remove the following flags:
...
+
+Storybook 7 will prepare manager-code for the browser using ESbuild (before it was using a combination of webpack + babel).
+This is a very important change, though it will not affect most addons.
+It means that when creating custom addons, particularly custom addons within the repo in which they are consumed,
+you will need to be aware that this code is not passed though babel, and thus will not use your babel config.
+
```
Comment (shilman) [inline MIGRATION.md:565]:
```suggestion
```

File: MIGRATION.md [hunk 6]
```diff
@@ -494,6 +497,112 @@ The new CLI commands remove the following flags:
...
+If this is a problem for your addon, you need to pre-compile your addon's manager code to ensure it works.
+
+If you addon also introduces preview code (such a decorators) it will be passed though whatever builder + config the user has configured for their project; this hasn't changed.
+
+In both the preview and manager code it's good to remember [Storybook now targets modern browser only](#modern-browser-support).
+
+The package `@storybook/components` contain a lot of components useful for building addons.
+Some of these addons have been moved to a new package `@storybook/blocks`.
+These components were moved: `ColorControl`, `ColorPalette`, `ArgsTable`, `ArgRow`, `TabbedArgsTable`, `SectionRow`, `Source`, `Code` to `@storybook/blocks`.
```
Comment (shilman) [inline MIGRATION.md:582]:
```suggestion
These components were moved: `ColorControl`, `ColorPalette`, `ArgsTable`, `ArgRow`, `TabbedArgsTable`, `SectionRow`, `Source`, `Code`.
```

File: MIGRATION.md [hunk 7]
```diff
@@ -494,6 +497,112 @@ The new CLI commands remove the following flags:
...
+But if you're using an addon that hasn't been updated yet, it might not work.
+
+It's possible for example for older addons to use APIs that are no longer available in Storybook 7. 
+Your addon might not show upside of the storybook (manager) UI, or storybook might fail to start entirely.
+
+When this happens to you please open an issue on the addon's repo, and ask the addon author to update their addon to be compatible with Storybook 7.
+It's also useful for the storybook team to know which addons are not yet compatible, so please open an issue on the storybook repo as well; particularly if the addon is popular and causes a critical failure.
+
+Here's a list of popular addons that are known not to be compatible with Storybook 7 yet:
+- [ ] [storybook-addon-jsx](https://example.com)
```
Comment (shilman) [inline MIGRATION.md:598]: Update links

File: docs/migration-guide.md [hunk 8]
```diff
@@ -11,6 +11,7 @@ The rest of this guide will help you upgrade successfully, either automatically
 - [Webpack4 support discontinued](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#webpack4-support-discontinued)
 - [IE11 support discontinued](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#modern-browser-support)
 - [Minimum supported version of node is 16](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#dropped-support-for-node-15-and-below)
+- [Some addons might not work yet](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#specific-instructions-for-addon-users)
```
Comment (yannbf) [inline docs/migration-guide.md:14]:
```suggestion
- [Some community addons might not work yet](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#specific-instructions-for-addon-users)
```

File: MIGRATION.md [hunk 9]
```diff
@@ -494,6 +497,110 @@ The new CLI commands remove the following flags:
...
+Note that this before would import `addons` from `@storybook/addons`, but now it imports `{ addons }` from `@storybook/manager-api`.
+The `addons` export is now a named export only, there's no default export anymore, so make sure to update this usage.
+
+The package `@storybook/addons` is still available, but it's only for backwards compatibility. It's not recommended to use it anymore.
+
+It's also been used by addon creators to gain access to a few APIs like `makeDecorator`. 
+These APIs are now available in `@storybook/preview-api`.
+
+Storybook users have had access to a few storybook-lifecycle hooks such as `useChannel`, `useParameter`, `useStorybookState`;
+when these hooks are used in panels, they should be imported from `@storybook/manger-api`.
```
Comment (yannbf) [inline MIGRATION.md:525]:
```suggestion
when these hooks are used in panels, they should be imported from `@storybook/manager-api`.
```

File: MIGRATION.md [hunk 10]
```diff
@@ -31,7 +34,7 @@
     - [Addon-a11y: Removed deprecated withA11y decorator](#addon-a11y-removed-deprecated-witha11y-decorator)
   - [7.0 Vite changes](#70-vite-changes)
     - [Vite builder uses Vite config automatically](#vite-builder-uses-vite-config-automatically)
-    - [Vite cache moved to node_modules/.cache/.vite-storybook](#vite-cache-moved-to-node_modulescachevite-storybook)
+    - [Vite cache moved to node\_modules/.cache/.vite-storybook](#vite-cache-moved-to-node_modulescachevite-storybook)
```
Comment (shilman) [inline MIGRATION.md:37]:
```suggestion
    - [Vite cache moved to node_modules/.cache/.vite-storybook](#vite-cache-moved-to-node_modulescachevite-storybook)
```

File: MIGRATION.md [hunk 11]
```diff
@@ -77,7 +80,7 @@
     - [Dropped addon-docs manual babel configuration](#dropped-addon-docs-manual-babel-configuration)
     - [Dropped addon-docs manual configuration](#dropped-addon-docs-manual-configuration)
     - [Autoplay in docs](#autoplay-in-docs)
-    - [Removed STORYBOOK_REACT_CLASSES global](#removed-storybook_react_classes-global)
+    - [Removed STORYBOOK\_REACT\_CLASSES global](#removed-storybook_react_classes-global)
```
Comment (shilman) [inline MIGRATION.md:83]:
```suggestion
    - [Removed STORYBOOK_REACT_CLASSES global](#removed-storybook_react_classes-global)
```

**Review by shilman — APPROVED** [review]

**Pull request merged**